### ISSUE TITLE
[ENH]: Should pickling a CallbackRegistry with unpicklable callbacks emit a warning?

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#28602]
### Problem

Currently, pickling a CallbackRegistry silently drops registered callbacks, except those register using the internal `_connect_picklable` (which we are careful to call only with picklable callbacks).
- Should we emit a warning when a (user) registered-callback? cf #22482.
- Should we expose _connect_picklable?

**Comment by timhoffm** [comment]
We should not silently drop. I don't oversee what exposing _connect_picklable would imply. Warn is the safe minimal improvement we can make. When users are aware of the dropped callback, they can file a feature request if they want a bette solution.

**Comment by tacaswell** [comment]
I agree, we should start with warning and see what feature requests we get.
